plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/CodeGenerationState.kt

```diff
@@ -31,11 +33,13 @@ class CodeGenerationState(
     override var approach: String,
     val config: SessionStateConfig,
     val uploadId: String,
-    val currentIteration: Int,
+    override var currentIteration: Int? = 0,
```

Comment:
still unsure why we're allowing null here

Reply:
Var-property type is Int, which is not a type of overridden